solv/tdnfpackage.c

```diff
@@ -1655,24 +1660,22 @@ SolvReportProblems(
         type = solver_ruleinfo(pSolv, dwProblemId,
                                &dwSource, &dwTarget, &dwDep);
 
-        if (SkipBasedOnType(type, dwSkipProblem))
+        if ((dwSkipProblem != SKIPPROBLEM_NONE) &&
+            SkipBasedOnType(type, dwSkipProblem))
         {
             continue;
         }
 
         pszProblem = solver_problemruleinfo2str(pSolv, type, dwSource,
                                                 dwTarget, dwDep);
 
-        if (type == SOLVER_RULE_PKG_REQUIRES)
+        if (dwSkipProblem != SKIPPROBLEM_NONE &&
```

Comment:
How does SOLVER_RULE_PKG_REQUIRES type is related to SKIPPROBLEM_CONFLICTS and OBSOLETES?
Why are we adding dwSkipProblem != SKIPPROBLEM_NONE check?
As per my understanding, If --skip conflicts and obsolete are not provided then dwSkipProblem should be SKIPPROBLEM_NONE and type is SOLVER_RULE_PKG_REQUIRES, in that case also, flow should go inside if condition.

Reply:
This is not related to skipping conflicts or obsoletes. TDNF reports few false positives saying 'none of the providers can be installed'. For that I added one more check to confirm whether a provider is really not available.

Yes, with 'tdnf check' dwSkipProblem value will be NONE and we don't check for providers. That way we will report everything (including false positives), if there are any spec file issues we can address them with this method.
Or else we will end up blocking everything related to providers.

```
Check: pytests/tests/test_check_skip.py
def test_dummy_check(utils):
    ret = utils.run(['tdnf', 'check'])
    assert ' obsoletes ' in ret['stderr'][0]
    assert ' conflicts ' in ret['stderr'][1]
    assert ' provides ' in ret['stderr'][2]
```

Reply:
As discussed, This is just a work around to skip check_for_providers().
Actual issue need to be fixed  in  check_for_providers().